src/types/express-serve-static-core.ts

```diff
@@ -0,0 +1,9 @@
+import "express-serve-static-core";
+
+declare module "express-serve-static-core" {
+  interface Request {
+    user?: {
+      userId: string;
```

Comment:
Inside User object the id will be obviously user's ID. So probably just `id` is enough.